### PR TITLE
Fixes sprintf javascript error and adds new tests

### PR DESCRIPTION
**Pull request by shubhsingh594**
resolves issues from https://github.com/learningequality/ka-lite/pull/4861
@benjaoming  this is still giving me a time out, i have resolved all other issues.

### CODE REVIEW

**Comment by shubhsingh594** [comment]
@benjaoming can you rebuild it, and check if it passes.

**Comment by benjaoming** [comment]
@shubhsingh594 try merging #4891 and see if it makes a difference. A lot of PRs are unmerged atm.

**Comment by benjaoming** [comment]
The error in the test build is the same as the other times, no reason to do a rebuild, it's persistent for sure.

**Comment by shubhsingh594** [comment]
Fixes https://github.com/learningequality/ka-lite/issues/4785
Fixes https://github.com/learningequality/ka-lite/issues/4841

File: kalite/distributed/features/steps/learn_content.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 from kalite.testing.behave_helpers import build_url, reverse, alert_in_page, find_css_class_with_wait, \
     assert_no_element_by_css_selector
 
-TIMEOUT = 3
+TIMEOUT = 30
```
Comment (benjaoming) [inline kalite/distributed/features/steps/learn_content.py]: I think 30 seconds timeout mixed with `assert_no_element_by_css_selector` sounds slow, but I'll create a separate issue for it.

You are more than welcome to work on #4900 if you feel like it :) You don't need to solve all of it in one PR, but any changes that can help us optimize testing are very welcomed!!

**Comment by benjaoming** [comment]
Thanks for this, it seems that increasing the timeout did the trick. I opened up #4900 to investigate it further.

**Comment by benjaoming** [comment]
Btw. please setup your editor to help you adhere to pep8.

**Pull request merged**